### PR TITLE
Fix shell script tests

### PR DESCRIPTION
**Pull request by hoegaarden**
run them as either:
- `make test-sh`
- individually as e.g. `./lib/gitlib_test.sh`
- `bazel test //lib/...`

/cc @pswica @justaugustus

### CODE REVIEW

File: lib/gitlib.sh [hunk 1]
```diff
@@ -134,7 +134,7 @@ gitlib::github_api_token () {
 # returns 1 on failure
 # Disable shellcheck for dynamically defined variable
 # shellcheck disable=SC2154
-export PROGSTEP[gitlib::github_acls]="CHECK GITHUB AUTH"
```
Comment (pswica) [inline lib/gitlib.sh]: I think shellcheck is complaining about this. May just need to disable it on this line

**Comment by pswica** [comment]
/hold
I'll put a hold in case we are waiting for that 12 release
@justaugustus

**Comment by pswica** [comment]
/lgtm
brilliant!

**Comment by justaugustus** [comment]
@hoegaarden -- can you rebase this?

File: lib/gitlib_test.sh [hunk 2]
```diff
@@ -8,6 +23,9 @@ source $TOOL_LIB_PATH/releaselib.sh
 
 readonly TESTDATA="$( cd "$(dirname "$0")" && pwd )/testdata"
 
+set -e
```
Comment (justaugustus) [inline lib/gitlib_test.sh]: Prefer:
```
set -o errexit
set -o nounset
set -o pipefail
```

and set options immediately after file header.

File: lib/releaselib_test.sh [hunk 3]
```diff
@@ -1,26 +1,54 @@
...
+# shellcheck source=./lib/common.sh
+source "$(dirname "$(readlink -ne "${BASH_SOURCE[0]}")")/common.sh"
+# shellcheck source=./lib/testing.sh
+source "${TOOL_LIB_PATH}/testing.sh"
+# shellcheck source=./lib/gitlib.sh
+source "${TOOL_LIB_PATH}/gitlib.sh"
+# shellcheck source=./lib/releaselib.sh
+source "${TOOL_LIB_PATH}/releaselib.sh"
+
+set -e
```
Comment (justaugustus) [inline lib/releaselib_test.sh]: Prefer:
```
set -o errexit
set -o nounset
set -o pipefail
```

and set options immediately after file header.

File: lib/releaselib_test.sh [hunk 4]
```diff
@@ -54,35 +82,36 @@ read -r -d '' data <<'EOF' || true
...
-rm -f $published_file
+  ##############################################################################
+  # END TESTING release::gcs::verify_latest_update()
+  ##############################################################################
+}
 
-##############################################################################
-# END TESTING release::gcs::verify_latest_update()
-##############################################################################
+testMain "$@"
```
Comment (justaugustus) [inline lib/releaselib_test.sh]:
```suggestion
test_main "$@"
```

File: lib/gitlib_test.sh (hunk 2 above)
Comment (hoegaarden) [reply lib/gitlib_test.sh]: Yeah, we can do that when all the things we source ( == things we want to test and they depend on [`gitlib.sh`, `common.sh`, `releaselib.sh`]) don't run code that relies on undefined variables and doesn't ignore errors and all that jazz. When can we do that? When we have tests in place. I guess you see where I am going here ...

File: lib/releaselib_test.sh (hunk 3 above)
Comment (hoegaarden) [reply lib/releaselib_test.sh]: https://github.com/kubernetes/release/pull/798#discussion_r307699214

File: lib/releaselib_test.sh (hunk 4 above)
Comment (hoegaarden) [reply lib/releaselib_test.sh]: ~Why?~ fiiine ...

**Pull request merged**